### PR TITLE
SPEC-962: Remove support for MongoDB 2.4 in CRUD spec

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/SPEC-962

https://github.com/mongodb/specifications/pull/207 for [SPEC-836](https://jira.mongodb.org/browse/SPEC-836) removed the useCursor option in AggregateOptions, which is related to drivers dropping support for 2.4 servers.

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
Can you also remove the `*pre_2.6*` tests?
```
$ find . -name "*pre*.yml"
./source/crud/tests/write/findOneAndReplace-upsert_pre_2.6.yml
./source/crud/tests/write/replaceOne-pre_2.6.yml
./source/crud/tests/write/updateMany-pre_2.6.yml
./source/crud/tests/write/updateOne-pre_2.6.yml
```

File: source/crud/crud.rst [hunk 1]
```diff
@@ -189,7 +189,6 @@ Read
      * The number of documents to return per batch.
      *
      * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
-     * For servers < 2.6, this option is ignored and should not be sent as aggregation cursors are not available.
```
Comment (jmikola) [inline source/crud/crud.rst]: This likely should have been removed in https://github.com/mongodb/specifications/commit/7c5344205da658450228d2cd5b93f67f3c5c930e#diff-5efd2e4ce61f4ccda0ce031099303996 for [SPEC-836](jira.mongodb.org/browse/SPEC-836).

File: source/crud/crud.rst [hunk 2]
```diff
@@ -1540,7 +1538,7 @@ Q: Where is write concern?
   However, it might be that a driver needs to expose write concern to a user per operation for various reasons. As noted before, it is permitted to specify this, along with other driver-specific options, in some alternative way.
 
 Q: How do I throttle unacknowledged writes now that write concern is longer defined on a per operation basis?
-  Some users used to throttle unacknowledged writes by using a write concern every X number of operations. The proper way to handle this on >= 2.6 servers is to use the bulk write API. Users working with servers < 2.6 should manually send a ``getLastError`` command every X number of operations if the driver does not support write concerns per operation.
+  Some users used to throttle unacknowledged writes by using an acknowledged write concern every X number of operations. Going forward, the proper way to handle this is by using the bulk write API.
```
Comment (jmikola) [inline source/crud/crud.rst]: I was comfortable removing the last sentence in the original text, as that approach is exactly what is specified by the first sentence. If users decide they still want to use unacknowledged writes, for which the CRUD spec will allow legacy opcodes, they can still periodically use an acknowledged write concern. That will effectively interleave their legacy opcodes with write commands on 2.6+ and achieve the same throttling.

File: source/crud/crud.rst [hunk 3]
```diff
@@ -1540,7 +1538,7 @@ Q: Where is write concern?
   However, it might be that a driver needs to expose write concern to a user per operation for various reasons. As noted before, it is permitted to specify this, along with other driver-specific options, in some alternative way.
 
 Q: How do I throttle unacknowledged writes now that write concern is longer defined on a per operation basis?
```
Comment (ShaneHarvey) [inline source/crud/crud.rst]: is **no** longer?

Comment (jmikola) [reply source/crud/crud.rst]: Fixed in https://github.com/mongodb/specifications/pull/208/commits/5d2cc4c5887c911d4d00b5c6fadf2dcd3a89fe7a

**Comment by ajdavis** [comment]
I abstain.

**Pull request merged**